src/sctools/gtf.py

```diff
@@ -260,6 +261,46 @@ def _resolve_multiple_gene_names(gene_name: str):
     )
 
 
+def get_mitochondrial_gene_names(
+    files: Union[str, List[str]] = "-", mode: str = "r", header_comment_char: str = "#"
+) -> Set[str]:
+    """Extract mitocholdrial gene names from GTF file(s) and returns a set of mitochondrial
+     gene id occurrence in the given file(s).
+
+    Parameters
+    ----------
+    files : Union[str, List], optional
+        File(s) to read. If '-', read sys.stdin (default = '-')
+    mode : {'r', 'rb'}, optional
+        Open mode. If 'r', read strings. If 'rb', read bytes (default = 'r').
+    header_comment_char : str, optional
+        lines beginning with this character are skipped (default = '#')
+
+    Returns
+    -------
+    Set(str)
+        A set of the mitochondrial gene ids
+    """
+
+    mitochondrial_gene_ids: Set[str] = set()
+    for record in Reader(files, mode, header_comment_char).filter(
+        retain_types=["gene"]
+    ):
+        gene_name = record.get_attribute("gene_name")
+        gene_id = record.get_attribute("gene_id")
+
+        if gene_name is None:
+            raise ValueError(
+                f"Malformed GTF file detected. Record is of type gene but does not have a "
+                f'"gene_name" field: {record}'
+            )
+        if re.match('^mt-', gene_name, re.IGNORECASE):
+            if gene_id not in mitochondrial_gene_ids:
```

Comment:
Do we need this explicit check? Since this is a set, you should be able to add and entry twice without duplication.